pycon/finaid/views.py

```diff
@@ -425,3 +425,33 @@ def get_value(name, object):
         writer.writerow(data)
 
     return response
+
+
+@login_required
+def receipt_form(request):
+    receipts = Receipt.objects.filter(user__username=request.user.username)
+
+    if request.method == 'POST':
+        form = ReceiptForm(request.POST, request.FILES)
+
+        if form.is_valid():
+            form.instance.user = request.user
+            form.instance.application = request.user.financial_aid
+            form.save()
+
+            # Display a message to user
+            messages.add_message(request, messages.INFO,
+                                _(u"Receipt submitted"))
+
+    else:
+        form = ReceiptForm()
+    return render(request, "finaid/receipt_form.html", {
+        'form': form,
+        'receipts': receipts
+    })
+
+
+@login_required
+def receipt_delete(request, pk):
```

Comment:
We'll want to update this if they do want to allow deleting receipts.

One of the web rules we try to follow is that GETs are always safe - they never modify data. To modify data, a POST is needed (or PUT or some other oddball methods).

So the pattern we usually follow for a delete view is that a GET displays a form asking if they really want to delete whatever it is, and if they submit the form, it POSTs to the same view, which then does the delete (and redirects to some other page).

Django has a generic class-based view that implements this very simply for you, if you want to start learning about class-based views.